tests/conftest.py

```diff
@@ -231,6 +232,62 @@ def _benchmark_time():
     return _benchmark_time()
 
 
+@pytest.fixture(scope="function")
+def benchmark_coiled_prometheus(test_run_benchmark):
+    """Benchmark the coiled prometheus metrics of executing some code.
+
+    Yields
+    ------
+    Context manager that records the coiled prometheus metrics of executing
+    the ``with`` statement if run as part of a benchmark, or does nothing otherwise.
+
+    Example
+    -------
+    .. code-block:: python
+
+        def test_something(benchmark_coiled_prometheus):
+            with benchmark_coiled_prometheus:
+                do_something()
+    """
+
+    @contextlib.contextmanager
+    def _benchmark_coiled_prometheus(client):
+        if not test_run_benchmark:
+            yield
+        else:
+            start = math.floor(time.time())
+            yield
+            end = math.ceil(time.time())
+            cluster = client.cluster
+            test_run_benchmark.scheduler_memory_max = cluster.get_aggregated_metric(
+                query="(host_memory_total-host_memory_available)&scheduler",
```

Comment:
```suggestion
                query="scheduler:(host_memory_total-host_memory_available)&scheduler",
```

`&scheduler` is a filter (like `&workers-up-30s`), `scheduler:` says that we want scheduler rather than worker metrics.

Example at https://github.com/coiled/platform/blob/e0913930dd7f04e618027176d28dbf9a4b9b394e/declarative/services/prometheus/tags.py#L38

Reply:
So I don't need the suffix `&scheduler` anymore, do I?

Reply:
I'd leave it in, it might not be needed in some specific cases (not sure about this one), but it's a way of addressing https://github.com/coiled/platform/issues/1358 in the cases where that applies.